### PR TITLE
Hiding sphinx-needs warnings during doc build

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

Skipping Sphinx-Needs warnings on public doc build.

## Why is the change being made?

There were 500 of these warnings, and they were obscuring the rest of the doc build.

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify any new/changed code.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/developer/tooling.html#add-release-notes) have been updated if necessary.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -212,6 +212,20 @@ def run(self):
             )
 
 
+class SkipNeedsDirective(Directive):
```
Comment (ntouran) [inline doc/conf.py:215]: probably worth adding docstring:

```
A no-op directive that filters out any sphinx-need directives from docs.

Temporary patch until we figure out a different/better way to maintain formal QA docs
```

**Comment by john-science** [comment]
@ntouran Please see the following GitHub Action log for proof my no-opt directive worked:

https://github.com/terrapower/armi/actions/runs/11520152570/job/32070797806?pr=1973#step:8:2700

File: .github/workflows/docs.yaml [hunk 2]
```diff
@@ -2,8 +2,8 @@ name: Documentation
 
 on:
   push:
-    branches:
-      - main
+#    branches:
+#      - main
```
Comment (john-science) [inline .github/workflows/docs.yaml:6]:
```suggestion
    branches:
      - main
```

File: .github/workflows/docs.yaml [hunk 3]
```diff
@@ -36,11 +36,12 @@ jobs:
...
-          folder: doc/_build/html
-          target-folder: armi
+#      - name: deploy
+#        uses: JamesIves/github-pages-deploy-action@v4.6.1
+#        with:
+#          token: ${{ secrets.ACCESS_TOKEN }}
+#          repository-name: ${{ github.repository_owner }}/terrapower.github.io
+#          branch: main
+#          folder: doc/_build/html
+#          target-folder: armi
```
Comment (john-science) [inline .github/workflows/docs.yaml:46]:
```suggestion
      - name: deploy
        uses: JamesIves/github-pages-deploy-action@v4.6.1
        with:
          token: ${{ secrets.ACCESS_TOKEN }}
          repository-name: ${{ github.repository_owner }}/terrapower.github.io
          branch: main
          folder: doc/_build/html
          target-folder: armi
```

File: doc/conf.py (hunk 1 above)
Comment (john-science) [inline doc/conf.py:215]:
```suggestion
class SkipNeedsDirective(Directive):
    """
    A no-op directive that filters out any sphinx-need directives from docs.

    Temporary patch until we figure out a different/better way to maintain formal QA docs.
    """
```

Comment (john-science) [reply doc/conf.py:215]: Done

File: .github/workflows/docs.yaml [hunk 4]
```diff
@@ -44,3 +44,4 @@ jobs:
           branch: main
           folder: doc/_build/html
           target-folder: armi
+
```
Comment (john-science) [inline .github/workflows/docs.yaml:47]:
```suggestion
```

**Review by drewj-tp — APPROVED** [review]

**Pull request merged**